Assignments/A2/Distro/MiniMax_search.c

```diff
@@ -275,8 +301,60 @@ double MiniMax(double gr[graph_size][4], int path[1][2],
   // Stub so that the code compiles/runs - This will be removed and replaced by
   // your code!
 
-  path[0][0] = mouse_loc[0][0];
-  path[0][1] = mouse_loc[0][1];
+  // TODO(@Sam): Figure out how assigning "path" works
+  // path[0][0] = mouse_loc[0][0];
+  // path[0][1] = mouse_loc[0][1];
+  Cord cord = {mouse_loc[0][0], mouse_loc[0][1]};
+  if (depth >= maxDepth) {
+    return utility(cat_loc, cheese_loc, mouse_loc, cats, cheeses, depth, gr);
+  } else if (agentId == 0) {
+    // Mouse - maximizing agent
+    double max_eval = DBL_MIN;
+    for (int direction = 0; direction < 4; ++direction) {
```

Comment:
a potential improvement to the minimax algo would be to order the moves from best to worst. in the case of the mouse, we could run a quick for loop over the actions and sort them in increasing distance to the closest cheese. 

similarly, we could do this for each cat and look at actions that are closer to the mouse first.

Reply:
we *could* but for the purpose of this assignment, I believe we're computing every possible continuation until our `maxDepth` is reached.

This means, the same computation would be done, regardless of search-ordering.

In face, if we were to try and re-order the moves considered, that would only add overhead.

Reply:
If we are performing alpha-beta pruning, I believe this would help prune branches more efficiently, as the agent is "most likely" to increase their utility with moves that are closer to their goal. I agree that it would add overhead, although if it helps prune branches faster I believe it will still result in efficiency gains.

The ordering by euclidean distance is not a complex measure to calculate either, and it would only increase complexity by an order of O(b), which would still be dominated by the overall complexity of O(b^m)

Reply:
Ah I see what you're saying - I'll add that to the PR before merging. Good point :)